File: bench/io/bench_readlines.nim
# Count lines and bytes, one line at a time. See bench_readlines.c for the
# result fold. readLine strips the terminator, so it is added back.
import std/[times, strutils]

let t0 = epochTime()
var lines: int64 = 0
var total: int64 = 0
var f: File
if not open(f, "io_input.txt"):
  quit(1)
var line = ""
while f.readLine(line):
  lines += 1
  total += line.len + 1
close(f)
let ms = int((epochTime() - t0) * 1000)
echo "result: ", lines * 100000000 + total
echo "elapsed: ", ms

File: bench/io/bench_readlines.lua
-- Count lines and bytes, one line at a time. See bench_readlines.c for the
-- result fold. io.lines strips the terminator, so it is added back.
local t0 = os.clock()
local lines, total = 0, 0
for line in io.lines("io_input.txt") do
    lines = lines + 1
    total = total + #line + 1
end
local ms = (os.clock() - t0) * 1000
print(string.format("result: %.0f", lines * 100000000 + total))
print(string.format("elapsed: %.0f", ms))
